### PR TITLE
feat: MLIBZ-2362 Implement Register/Unregister methods

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
Implement Register/Unregister methods

#### Changes
Added register/unregister methods to User.

#### Tests
Instrumented

### CODE REVIEW

File: android-lib/src/main/java/com/kinvey/android/Client.java [hunk 1]
```diff
@@ -1064,5 +1064,10 @@ public void stopKinveyHandlerThread() {
             kinveyHandlerThread.interrupt();
         }
     }
+
+    @Override
+    public String getDeviceId() {
+        return new UdidFactory(context).getDeviceUuid().toString();
```
Comment (vinaygahlawat) [inline android-lib/src/main/java/com/kinvey/android/Client.java]: If we can, we should fix the misspelling of `UdidFactory`.

Comment (yuliya-guseva) [reply android-lib/src/main/java/com/kinvey/android/Client.java]: This is internal class, we can. Updated.

File: java-api-core/src/com/kinvey/java/store/BaseUserStore.java [hunk 2]
```diff
@@ -143,6 +145,42 @@ public static boolean exists( String username, AbstractClient client) throws IOE
...
+
+    /**
+     * Unregister the active user from realtime messaging.
+     * @throws IOException
+     */
+    public static void unRegisterRealtime() throws IOException {
+        if (AbstractClient.sharedInstance().getActiveUser() == null) {
+            throw new KinveyException("User object has to be the active user in order to register for realtime messages");
+        }
+        RealtimeRouter.getInstance().unInitialize();
```
[the remark this answers is not in the document]
Comment (yuliya-guseva) [reply java-api-core/src/com/kinvey/java/store/BaseUserStore.java]: If the RealtimeRouter is not initialized, router won't initialize him to uninitialize. The Router's methods has initialize checking before executing. But I've added additianal isInitialize checking to BaseUserStore to don't make a unnessesary request to the server.

**Comment by yuliya-guseva** [comment]
Currently we have one small test for register/unregister. DeviceId already is kept in the SharedPreferences. 
Added small device uuid test.

**Pull request merged**